app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/state/MessageComposerState.kt

```diff
@@ -342,6 +325,28 @@ data class MessageComposerInnerState(
     fun cancelReply() {
         quotedMessageData = null
     }
+
+    fun specifySelfDeletionTime(selfDeletionDuration: SelfDeletionDuration) {
```

Comment:
The `SelfDeletingMessage` type is set only by this method, right? So it means, that this option is not stored anywhere (in scala app it looks like the self-deleting is stored for each conversation so after you kill the app it still remembers last chosen option for the given conversation) but here it won't be remembered even if the user simply closes the keyboard, because then the app changes the input state to `Inactive` and after that, when user clicks on the input to write another message, then it will be changed to `Active` with type `NewMessage`. 

Is that intentional? Or maybe it will be covered later, I just wanted to raise awareness to have it discussed with Wolfgang as it would make a better UX. 😄

Reply:
This is going to be tackeld later indeed, thanks for bringing this up